api/app/src/sequelize/migrations/2023050403101805-update-patient-data.ts

```diff
@@ -0,0 +1,70 @@
+import { DataTypes } from "sequelize";
+import type { Migration } from "..";
+import { PatientModel } from "../../models/medical/patient";
+
+// Use 'Promise.all' when changes are independent of each other
+// Docs: https://sequelize.org/api/v6/class/src/dialects/abstract/query-interface.js~queryinterface
+export const up: Migration = async ({ context: queryInterface }) => {
+  return queryInterface.sequelize.transaction(async transaction => {
+    const patients = (await queryInterface.select(null, PatientModel.NAME, {
+      transaction,
+    })) as any[]; //eslint-disable-line @typescript-eslint/no-explicit-any
+    const newPatients: any[] = []; //eslint-disable-line @typescript-eslint/no-explicit-any
+    for (const patient of patients) {
+      let newPatientData = { ...patient.data };
+      if (newPatientData.address && !Array.isArray(newPatientData.address)) {
+        newPatientData = { ...newPatientData, address: [newPatientData.address] };
+      }
+      if (
+        newPatientData.contact &&
+        !Array.isArray(newPatientData.contact) &&
+        (newPatientData.contact.email || newPatientData.contact.phone)
+      ) {
+        newPatientData = { ...newPatientData, contact: [newPatientData.contact] };
+      }
+      newPatients.push({ ...patient, data: { ...newPatientData } });
+    }
+    await queryInterface.bulkDelete(PatientModel.NAME, {}, { transaction });
```

Comment:
Can't we update the patient data instead of deleting/inserting the records?

Reply:
bulkUpdate() doesn't seem to work nicely - I've tried several workarounds to no avail.

to avoid burning more time here will leave as-is - there's not a huge amount of records to update, and there's no risk of data loss since everything is under a txn